File: Business/eUniversity.Business.ViewModels/Curriculum/CurriculumHeaderViewModel.cs
using System;

namespace eUniversity.Business.ViewModels.Curriculum
{
    /// <summary>
    /// View model for curriculum header servition
    /// </summary>
    public class CurriculumHeaderViewModel
    {
        public long Id { get; set; }

        public SelectedItemViewModel Specialization{get;set;}

        public SelectedItemViewModel CountSemesters { get; set; }

        public DateTime DateOfEnactment
        {
            get
            {
                if (string.IsNullOrEmpty(DateOfEnactmentDisplay))
                {
                    DateOfEnactmentDisplay = DateTime.Today.ToShortDateString();
                }
                return DateTime.Parse(DateOfEnactmentDisplay);
            }
            set
            {
                DateOfEnactmentDisplay = value.ToShortDateString();
            }
        }

        public string DateOfEnactmentDisplay { get; set; }
    }
}

File: Web/eUniversity.Web/Scripts/viewModels/curriculum/curriculumHeaderViewModel.js
﻿function CurriculumHeaderViewModel(serverModel) {
    var self = this;
    self.IsActive = ko.observable(true);
    self.IsCompleted = ko.observable(false);

    self.Specialization = ko.observable();
    self.CountSemesters = ko.observable();
    self.DateOfEnactmentDisplay = ko.observable();

    ko.mapping.fromJS(serverModel, {}, self);
}